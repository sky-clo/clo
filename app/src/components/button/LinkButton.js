import React from "react";

export default function LinkButton(props) {
  const toolkitClass =
    props.primary === false ? "c-btn--secondary" : "c-btn--primary";
  const className = `c-btn ${toolkitClass} ${props.className}`;

  return (
    <a href={props.href} className={className}>
      {props.children}
    </a>
  );
}
