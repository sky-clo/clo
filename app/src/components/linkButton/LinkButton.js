import React from "react";

export default function LinkButton({
  children,
  className,
  href,
  primary,
  ...props
}) {
  const toolkitClass =
    primary === false ? "c-btn--secondary" : "c-btn--primary";

  return (
    <a href={href} className={`c-btn ${toolkitClass} ${className}`} {...props}>
      {children}
    </a>
  );
}
