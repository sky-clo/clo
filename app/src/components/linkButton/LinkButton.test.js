import React from "react";
import { shallow } from "enzyme";

import LinkButton from "./LinkButton";

it("LinkButton should render a primary button", () => {
  const render = shallow(
    <LinkButton href="/" className="test" data-test="my-button">
      My Button
    </LinkButton>
  );
  expect(render).toMatchSnapshot();
});

it("LinkButton should render a secondary button", () => {
  const render = shallow(
    <LinkButton href="/" primary={false} className="test" data-test="my-button">
      My Button
    </LinkButton>
  );
  expect(render).toMatchSnapshot();
});
