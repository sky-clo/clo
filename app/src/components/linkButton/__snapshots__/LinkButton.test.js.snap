// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton should render a primary button 1`] = `
<a
  className="c-btn c-btn--primary test"
  data-test="my-button"
  href="/"
>
  My Button
</a>
`;

exports[`LinkButton should render a secondary button 1`] = `
<a
  className="c-btn c-btn--secondary test"
  data-test="my-button"
  href="/"
>
  My Button
</a>
`;
